components/FileUploadField.tsx

```diff
@@ -0,0 +1,50 @@
+import { AttachmentIcon } from '@chakra-ui/icons';
+import { Button } from '@chakra-ui/react';
+import { SyntheticEvent, useRef } from 'react';
+
+// File upload button props
+type FileUploadProps = {
+  currentFile: File | null; // currently uploaded file
+  onUploadFile: (selectedFile: File) => void; // handle file upload
+};
+
+//TODO: rename this better??
+/**
+ * File upload button component allowing users to upload one file
+ * @param props - props
+ * @returns File upload button component allowing users to upload one file
+ */
+export default function FileUpload(props: FileUploadProps) {
+  const hiddenFileInput = useRef<HTMLInputElement>(null);
+
+  const handleFileUpload = () => {
```

Comment:
I would rename this to something like `handleClick` as it's more descriptive of the event being handled